cranelift/filetests/filetests/runtests/bextend.clif

```diff
@@ -5,85 +5,52 @@ target x86_64
 target s390x
 target riscv64
 
-function %bextend_b1_b8(b1) -> b8 {
-block0(v0: b1):
-  v1 = bextend.b8 v0
+function %bextend_i8_i16(i8) -> i16 {
+block0(v0: i8):
+  v1 = bextend.i16 v0
   return v1
 }
-; run: %bextend_b1_b8(true) == true
-; run: %bextend_b1_b8(false) == false
+; run: %bextend_i8_i16(1) == 1
+; run: %bextend_i8_i16(0) == 0
 
-function %bextend_b1_b16(b1) -> b16 {
-block0(v0: b1):
-  v1 = bextend.b16 v0
+function %bextend_i8_i32(i8) -> i32 {
+block0(v0: i8):
+  v1 = bextend.i32 v0
   return v1
 }
-; run: %bextend_b1_b16(true) == true
-; run: %bextend_b1_b16(false) == false
+; run: %bextend_i8_i32(1) == 1
+; run: %bextend_i8_i32(0) == 0
 
-function %bextend_b1_b32(b1) -> b32 {
-block0(v0: b1):
-  v1 = bextend.b32 v0
+function %bextend_i8_i64(i8) -> i64 {
+block0(v0: i8):
+  v1 = bextend.i64 v0
   return v1
 }
-; run: %bextend_b1_b32(true) == true
-; run: %bextend_b1_b32(false) == false
-
-function %bextend_b1_b64(b1) -> b64 {
-block0(v0: b1):
-  v1 = bextend.b64 v0
-  return v1
-}
-; run: %bextend_b1_b64(true) == true
-; run: %bextend_b1_b64(false) == false
-
-
-function %bextend_b8_b16(b8) -> b16 {
-block0(v0: b8):
-  v1 = bextend.b16 v0
-  return v1
-}
-; run: %bextend_b8_b16(true) == true
-; run: %bextend_b8_b16(false) == false
-
-function %bextend_b8_b32(b8) -> b32 {
-block0(v0: b8):
-  v1 = bextend.b32 v0
-  return v1
-}
-; run: %bextend_b8_b32(true) == true
-; run: %bextend_b8_b32(false) == false
-
-function %bextend_b8_b64(b8) -> b64 {
-block0(v0: b8):
-  v1 = bextend.b64 v0
-  return v1
-}
-; run: %bextend_b8_b64(true) == true
-; run: %bextend_b8_b64(false) == false
+; run: %bextend_i8_i64(1) == 1
+; run: %bextend_i8_i64(0) == 0
 
 
-function %bextend_b16_b32(b16) -> b32 {
-block0(v0: b16):
-  v1 = bextend.b32 v0
+function %bextend_i16_i32(i16) -> i32 {
+block0(v0: i16):
+  v1 = bextend.i32 v0
   return v1
```

Comment:
Can bextend be replaced with uextend (for non-vector bools) and sextend (for vector bools)?

Reply:
I'd be in favor of removing `bextend` and `breduce`. The latter can also be replaced by `band_imm`.

Reply:
Yep, if we no longer have bools then these instructions no longer make sense; the user can zero- or sign-extend an integer (which happens to be a 0/1 or 0/-1 "truthy" value) in the usual ways.

Reply:
I really like using `uextend`/`sextend` in place of `bextend`, and have made that refactoring. I've also removed `bconst`, `bextend`, `breduce` and `bint`, which has collapsed a lot of cases in the test suite.